### PR TITLE
Minor Performance Fixes

### PR DESCRIPTION
**Pull request by xrobau**
## Description
This consists of two commits - the first slightly reorders the way the query is generated so it puts less load on the database.  The second disables pythons Garbage Collector when building lists. This has little effect on newer python versions, but a dramatic effect on older ones (eg, Ubuntu 14.04)

## Motivation and Context
This was written because of suboptimal queries and performance

## How Has This Been Tested?
Has been running in production for several weeks

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/models.py [hunk 1]
```diff
@@ -92,10 +92,13 @@ def get_active(swLat, swLng, neLat, neLng):
...
                      .where((Pokemon.disappear_time > datetime.utcnow()) &
-                            (Pokemon.latitude >= swLat) &
-                            (Pokemon.longitude >= swLng) &
-                            (Pokemon.latitude <= neLat) &
-                            (Pokemon.longitude <= neLng))
+                            (((Pokemon.latitude >= swLat) &
+                              (Pokemon.longitude >= swLng) &
+                              (Pokemon.latitude <= neLat) &
+                              (Pokemon.longitude <= neLng)) |
+                             (Pokemon.pokemon_id << Pokemon.rare_pokemon)
```
Comment (xrobau) [inline pogom/models.py]: Whoops, that's for another PR, need to fix.

File: pogom/models.py [hunk 2]
```diff
@@ -183,9 +203,17 @@ def get_appearances(cls, pokemon_id, last_appearance):
                  .order_by(Pokemon.disappear_time.asc())
                  .dicts()
                  )
+
+        # Performance: Disable the garbage collector prior to creating a (potentially) large dict with append()
+        gc.disable()
```
Comment (Xcelled) [inline pogom/models.py]: I'd take this out and just do a `return list(query)`. No sense in doing things the hard and slow way for no reason.

File: pogom/models.py [hunk 3]
```diff
@@ -331,10 +365,16 @@ def get_gyms(swLat, swLng, neLat, neLng):
                             (Gym.longitude <= neLng))
                      .dicts())
 
+        # Performance: Disable the garbage collector prior to creating a (potentially) large dict with append()
+        gc.disable()
```
Comment (Xcelled) [inline pogom/models.py]: Likewise ^

File: pogom/models.py [hunk 4]
```diff
@@ -358,10 +398,16 @@ def get_recent(swLat, swLng, neLat, neLng):
                         (ScannedLocation.longitude <= neLng))
                  .dicts())
 
+        # Disable the garbage collector prior to creating a (potentially) large dict with append()
+        gc.disable()
```
Comment (Xcelled) [inline pogom/models.py]: And again^

File: pogom/models.py [hunk 5]
```diff
@@ -183,9 +203,17 @@ def get_appearances(cls, pokemon_id, last_appearance):
                  .order_by(Pokemon.disappear_time.asc())
                  .dicts()
                  )
+
+        # Performance: Disable the garbage collector prior to creating a (potentially) large dict with append()
+        gc.disable()
+
```
Comment (Infernoman) [inline pogom/models.py]: If im not mistaken this one too ^

**Comment by Xcelled** [comment]
Github seems to be messing up the line comments. Anything that uses append with no modifications should be converted to `return list(query)`.

**Comment by Infernoman** [comment]
wow are they ever. will be a +1 from me after the changes @Xcelled spoke about.

File: pogom/models.py (hunk 5 above)
Comment (shivampaw) [inline pogom/models.py]: Aka the first one that was pointed out ^^

**Comment by Infernoman** [comment]
@Xcelled the stats page is going to need to be handled different unless you leave the iteration and addition to a new array.

**Comment by Xcelled** [comment]
@Infernoman I didn't mark any lines that do anything except `query = blah; result = []; for x in query: result.append(x); return result;` Blame github.

**Comment by xrobau** [comment]
Thanks, @Xcelled, I didn't even think of just doing a list(). I'll redo this pull request today!

**Comment by Romfrosk** [comment]
Can't wait until this gets properly added :) 
Any performance boost is welcomed :)

**Comment by xrobau** [comment]
There we go, replaced with list() when I could, and it seemed noticably faster. I didn't do any performance testing against gc.disable()/gc.enable(), because either of them are faster than just a raw append()!

**Comment by Romfrosk** [comment]
Would you need to rebuild or drop any tables for this to be implemented?

**Comment by rehand** [comment]
I'd like to see this PR merged, would increase the overall performance on a Raspberry Pi 1st gen.

**Comment by Xcelled** [comment]
@Romfrosk No.

@rehand +1 me too.

@chuyskywalker this is a small change that shouldn't negatively affect anyone. Can we get it merged quickly?

**Pull request merged**